src/pkg/packager/creator/normal.go

```diff
@@ -119,6 +119,11 @@ func (pc *PackageCreator) LoadPackageDefinition(dst *layout.PackagePaths) (pkg t
 		}
 	}
 
+	// Perform early package validation.
+	if err := pkg.Validate(); err != nil {
+		return types.ZarfPackage{}, nil, fmt.Errorf("unable to validate package: %w", err)
+	}
+
 	return pkg, warnings, nil
```

Comment:
if the error returned from `pkg.Validate` is descriptive enough / does not require additional context, you can always:

```go
return pkg, warnings, pkg.Validate()
```

Reply:
Good point. I don't think the error returned from `Validate` needs extra context here so I changed to just return the `err` directly instead of wrapping.

I chose to leave the call to `pkg.Validate` the same rather than in a return statement because I want to ensure we always return an empty pkg on error because it's easier to test

Reply:
sounds good to me, feel free to resolve